homeassistant/components/sensor/london_air.py

```diff
@@ -0,0 +1,210 @@
+    'Islington',
+    'Kensington and Chelsea',
+    'Kingston',
+    'Lambeth',
+    'Lewisham',
+    'Merton',
+    'Redbridge',
+    'Richmond',
+    'Southwark',
+    'Sutton',
+    'Tower Hamlets',
+    'Wandsworth',
+    'Westminster']
+URL = ('http://api.erg.kcl.ac.uk/AirQuality/Hourly/'
+       'MonitoringIndex/GroupName=London/Json')
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(LOCATIONS):
+        vol.All(cv.ensure_list, [vol.In(list(AUTHORITIES))]),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Tube sensor."""
+    data = APIData()
+    data.update()
+    sensors = []
+    for name in config.get(LOCATIONS):
+        sensors.append(AirSensor(name, data))
+
+    add_devices(sensors, True)
+
+
+class APIData(object):
+    """Get the latest data for all authorities."""
+
+    def __init__(self):
+        """Initialize the AirData object."""
+        self.data = None
+
+    # Update only once in scan interval.
+    @Throttle(SCAN_INTERVAL)
+    def update(self):
+        """Get the latest data from TFL."""
+        response = requests.get(URL)
+        if response.status_code != 200:
+            _LOGGER.warning("Invalid response from API")
+        else:
+            self.data = parse_api_response(response.json())
+
+
+class AirSensor(Entity):
+    """Single authority air sensor."""
+
+    ICON = 'mdi:cloud-outline'
+
+    def __init__(self, name, APIdata):
+        """Initialize the sensor."""
+        self._name = name
+        self._APIdata = APIdata
+        self._site_data = None
+        self._state = None
+        self._updated = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def site_data(self):
+        """Return the dict of sites data."""
+        return self._site_data
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self.ICON
+
+    @property
+    def device_state_attributes(self):
+        """Return other details about the sensor state."""
+        attrs = {}
+        attrs['updated'] = self._updated
+        attrs['sites'] = len(self._site_data)
+        attrs['data'] = self._site_data
+        return attrs
+
+    def update(self):
+        """Update the sensor."""
+        self._APIdata.update()
+        self._site_data = self._APIdata.data[self._name]
+        self._updated = self._site_data[0]['updated']
+        sites_status = []
+        for site in self._site_data:
+            if site['pollutants_status'] != 'no_species_data':
+                sites_status.append(site['pollutants_status'])
+        if sites_status:
+            self._state = max(set(sites_status), key=sites_status.count)
+        else:
+            self._state = 'no_species_data'
+
+
+def parse_api_response(response):
```

Comment:
I think I will create a more fully fledged module for external use at a later date